src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,81 @@
+package duke.command;
+
+import duke.main.Ui;
+import duke.utils.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Delete command implementing command interface
+ */
+public class DeleteCommand implements Command {
+
+    /**
+     * attempt to delete task from task list and set ui response to the
+     * appropriate one
+     * @param task
+     * @param ui
+     * @param storage
+     * @param taskList
+     */
+    @Override
+    public void execute(String task, Ui ui, Storage storage, TaskList taskList) {
+
+        String[] token = task.split(" ", 2);
+        if (token.length < 2) {
+            ui.setResponse("Please specify which task to delete");
+            return;
+        }
+
+        String[] indices = token[1].split(" ");
```

Comment:
Agreed to the comment above, I think it would be better if you create an abstraction for parsing the user input and making sense of the user input, such as using a Parser class, so the code will be more readable if you put Parser.getFirstCommand, instead of token[1].slit(" ")